01/Task01.js

```diff
@@ -2,45 +2,51 @@ import React, { useState } from 'react';
 import Subject from './../src/libs/Subject';
 
 const Task01 = () => {
-    const [time, setTime] = useState(0);
-    const [tagName, setTagName] = useState('??');
-    const [cursorPosition, setCursorPosition] = useState('[?,?]');
-
-    const setEventTime = ({timeStamp}) => {
-        setTime( parseInt(timeStamp / 1000) );
-    }
-
-    const setEventTagName = ({target}) => {
-        setTagName(target.tagName);
-    }
-
-    const setEventCursorPosition = ({pageX, pageY}) => {
-        setCursorPosition(`[${pageX},${pageY}]`);
-    }
-
-    return (
-        <section>
-            <h1>Task 1</h1>
-            
-            <div onClick={ event => {
-                setEventTime(event);
-                setEventTagName(event);
-                setEventCursorPosition(event);
-            }}>
-                <p>
-                    <strong>Kliknij wybrany element:</strong> <a>link</a>, <button>button</button>, <span>span</span>
-                </p>
-
-
-                <ul>
-                    <li>time: { time }s</li>
-                    <li>tagName: { tagName }</li>
-                    <li>position: { cursorPosition }</li>
-                </ul>
-            </div>
-        </section>
-    )
-}
+  const [time, setTime] = useState(0);
+  const [tagName, setTagName] = useState('??');
+  const [cursorPosition, setCursorPosition] = useState('[?,?]');
+
+  const subject = new Subject();
+
+  const setEventTime = ({ timeStamp }) => {
+    setTime(parseInt(timeStamp / 1000));
+  };
+  subject.subscribe(setEventTime);
+
+  const setEventTagName = ({ target }) => {
+    setTagName(target.tagName);
+  };
+  subject.subscribe(setEventTagName);
+
+  const setEventCursorPosition = ({ pageX, pageY }) => {
+    setCursorPosition(`[${pageX},${pageY}]`);
+  };
+  subject.subscribe(setEventCursorPosition);
+
+  return (
+    <section>
+      <h1>Task 1</h1>
+
+      <div
+        onClick={(event) => {
+          setEventTime(event);
+          setEventTagName(event);
+          setEventCursorPosition(event);
```

Comment:
Po to zapisywaliśmy do `subject` funkcje, aby potem w tym miejscu tylko wywołać: `subject.notify(event)` :)